session/session.go

```diff
@@ -799,13 +800,18 @@ func (s *session) ExecRestrictedSQLWithSnapshot(sql string) ([]chunk.Row, []*ast
 	}
 	// Set snapshot.
 	if snapshot != 0 {
+		se.sessionVars.SnapshotInfoschema, err = domain.GetDomain(s).GetSnapshotInfoSchema(snapshot)
```

Comment:
Why not just use the snapshot infoschema in original session?

Reply:
If `s.sessionVars.SnapshotTS` is 0, it seems that we also need to do `GetSnapshotInfoSchema`.

Reply:
> it seems that we also need to do GetSnapshotInfoSchema

Why? Could you give a example?

Reply:
If the value of `snapshot` is `s.txn.StartTS()`, I think we need to call the `GetSnapshotInfoSchema`.